### PR TITLE
Update cryptography version to 3.2

### PR DESCRIPTION
**Pull request by blag**
Update `cryptography` to > 3.2 to avoid [CVE-2020-25659](https://github.com/pyca/cryptography/blob/master/CHANGELOG.rst#32---2020-10-25) and to fix the Travis-CI and CircleCI builds.

Closes #5065, #5066, and #5067.

### CODE REVIEW

**Review by armab — APPROVED** [review]
LGTM! Thanks for the maintenance work.

**Comment by blag** [comment]
Waiting on StackStorm/st2cd#446 and StackStorm/st2tests#198.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -25,6 +25,8 @@ Changed
 
   Contributed by @hnanchahal
 
+* Upgraded cryptography version to 3.2 to avoid CVE-2020-25659
```
Comment (armab) [inline CHANGELOG.rst:28]:
```suggestion
* Upgraded cryptography version to 3.2 to avoid CVE-2020-25659 (security) #5095
```

**Comment by nmaludy** [comment]
+1 this is needed to fix all of the exchange builds.

**Comment by armab** [comment]
As discussed before, merging to fix the Exchange CI.

**Pull request merged**